src/deadline/unreal_submitter/unreal_open_job/unreal_open_job_parameters_consistency.py

```diff
@@ -0,0 +1,268 @@
+        passed = True
+
+        missed_in_yaml, missed_in_data_asset = ParametersConsistencyChecker.symmetric_difference(
+            left=yaml_parameters, right=data_asset_parameters
+        )
+
+        if missed_in_yaml:
+            passed = False
+            missed_in_yaml_str = [f"{p[0]} ({p[1]})" for p in missed_in_yaml]
+            warn_message = "Data Asset's parameters missed in YAML: {}".format(
+                ", ".join(missed_in_yaml_str)
+            )
+            logger.warning(warn_message)
+            reasons.append(warn_message)
+
+        if missed_in_data_asset:
+            passed = False
+            missed_in_data_asset_str = [f"{p[0]} ({p[1]})" for p in missed_in_data_asset]
+            warn_message = "YAML's parameters missed in Data Asset: {}".format(
+                ", ".join(missed_in_data_asset_str)
+            )
+            logger.warning(warn_message)
+            reasons.append(warn_message)
+
+        reason = "\n".join(reasons) if not passed else "Parameters are consensual"
+
+        result = ParametersConsistencyCheckResult(passed, reason)
+
+        logger.info(result)
+
+        return result
+
+    @staticmethod
+    def fix_parameters_consistency(
+        missed_in_yaml: list[tuple[str, str]],
+        missed_in_data_asset: list[tuple[str, str]],
+        yaml_parameters: list[dict[str, Any]],
+        data_asset_parameters: list[dict[str, Any]],
+    ) -> list[dict[str, Any]]:
+
+        logger.info(f"Fixing missed parameters in YAML: {missed_in_yaml}")
+        logger.info(f"Fixing missed parameters in Data Asset: {missed_in_data_asset}")
+
+        if not (missed_in_yaml or missed_in_data_asset):
+            logger.info("No parameters to fix")
+            return yaml_parameters
+
+        fixed_parameters = []
+
+        for p in data_asset_parameters:
+            if (p["name"], p["type"]) not in missed_in_yaml:
+                fixed_parameters.append(p)
+        for p in yaml_parameters:
+            if (p["name"], p["type"]) in missed_in_data_asset:
+                fixed_param = {"name": p["name"], "type": p["type"]}
+                if "default" in p:
+                    fixed_param["default"] = p["default"]
+                fixed_parameters.append(p)
+
+        fixed_parameters.sort(key=lambda x: x["name"])
+
+        logger.info(f"Fixed parameters: {fixed_parameters}")
+
+        return fixed_parameters
+
+    @staticmethod
+    def fix_variables_consistency(
+        missed_in_yaml: list[tuple[str, str]],
+        missed_in_data_asset: list[tuple[str, str]],
+        yaml_variables: dict[str, str],
+        data_asset_variables: dict[str, str],
+    ) -> dict[str, str]:
+
+        logger.info(f"Fixing missed variables in YAML: {missed_in_yaml}")
+        logger.info(f"Fixing missed variables in Data Asset: {missed_in_data_asset}")
+
+        if not (missed_in_yaml or missed_in_data_asset):
+            logger.info("No variables to fix")
+            return yaml_variables
```

Comment:
We don't fix parameters automatically since user may want just to be notified it something is wrong with them. Therefore the notification about inconsistent parameters appers only when user add add/remove parameters from yaml and reopen the data  asset after it. 

Fix can't happen unless you call it in code directly or in cmd. But as a default behavior, we simply return a list of yaml parameters, because this is kind of source of truth

Reply:
Furthermore, if recieve only param name and type without value. So if there no missed parts, original lists is equal, so we can pick one of them and return it